src/lib/sidenav/sidenav.md

```diff
@@ -1,44 +1,133 @@
-| side | Sidenav appears _side-by-side_ with the primary content                                   |
+### Opening and closing a sidenav
 
-Using the `side` mode on mobile devices can affect the performance and is also not recommended by the
-[Material Design specification](https://material.io/guidelines/patterns/navigation-drawer.html#navigation-drawer-behavior).
+A `<mat-sidenav>` can be opened or closed using the `open()`, `close()` and `toggle()` methods. Each
+of these methods returns a `Promise<boolean>` that will be resolved with `true` when the sidenav
+finishes opening or `false` when it finishes closing.
 
-### Positioning the sidenav
-The `position` property determines whether the sidenav appears at the `"start"` or `"end"` of the
-container. This is affected by the current text direction ("ltr" or "rtl"). By default, the sidenav
-appears at the start of the container.
+The opened state can also be set via a property binding in the template using the `opened` property.
+The property supports 2-way binding.
 
+`<mat-sidenav>` also supports output properties for just open and just close events, The `(opened)`
+and `(closed)` properties respectively.
 
-### Sizing the sidenav
-The `<mat-sidenav>` will, by default, fit the size of its content. The width can be explicitly set
-via CSS:
+<!-- example(sidenav-open-close) -->
+
+All of these properties and methods work on `<mat-drawer>` as well.
+
+### Changing the sidenav's behavior
+
+The `<mat-sidenav>` can render in one of three different ways based on the `mode` property.
+
+| Mode   | Description                                                                             |
+|--------|-----------------------------------------------------------------------------------------|
+| `over` | Sidenav floats over the primary content, which is covered by a backdrop               |
+| `push` | Sidenav pushes the primary content out of its way, also covering it with a backdrop   |
+| `side` | Sidenav appears side-by-side with the main content, shrinking the main content's width to make space for the sidenav. |
+
+If no `mode` is specified, `over` is used by default.
+
+<!-- example(sidenav-mode) -->
+
+`<mat-drawer>` also supports all of these same modes.
+
+### Disabling automatic close
+
+Clicking on the backdrop or pressing the <kbd>Esc</kbd> key will normally close an open sidenav.
+However, this automatic closing behavior can be disabled by setting the `disableClose` property on
+the `<mat-sidenav>` or `<mat-drawer>` that you want to disable the behavior for.
+
+Custom handling for <kbd>Esc</kbd> can be done by adding a keydown listener to the `<mat-sidenav>`.
+Custom handling for backdrop clicks can be done via the `(backdropClick)` output property on
+`<mat-sidenav-container>`.
+
+<!-- example(sidenav-disable-close) -->
+
+### Setting the sidenav's size
+
+The `<mat-sidenav>` and `<mat-drawer>` will, by default, fit the size of its content. The width can
```

Comment:
Might need to mention that it will fit the size of its content just once - I've had issues where the sidenav fills in with items, then gets a scrollbar and the width changes but the sidenav doesn't update